cityehr-quick-start-guide/src/main/xslt/create-topic-html.xslt

```diff
@@ -25,15 +34,29 @@
         </xsl:call-template>
       </head>
       <body>
-        <div id="top-nav">
+        <nav id="top-nav">
           <xsl:apply-templates select="." mode="top-nav"/>
-        </div>
+        </nav>
+
+        <!-- Petal Edit Button -->
+        <section>
+          <xsl:variable name="petal-source-file-uri" select="com:document-uri(.)" />
+          <xsl:variable name="tokens" select="tokenize($petal-source-file-uri, '/')"/>
+          <xsl:variable name="orig-file-name" select="$tokens[last()]"/>
+          <xsl:variable name="petal-source-file" select="substring-after($petal-source-file-uri, $petal-github-repo-name || '/')" />
```

Comment:
We already have a function for obtaining the filename, you could replace:

```xslt
<xsl:variable name="petal-source-file-uri" select="com:document-uri(.)" />
<xsl:variable name="tokens" select="tokenize($petal-source-file-uri, '/')"/>
<xsl:variable name="orig-file-name" select="$tokens[last()]"/>
<xsl:variable name="petal-source-file" select="substring-after($petal-source-file-uri, $petal-github-repo-name || '/')" />
<xsl:variable name="petal-file-name" select="hcom:dita-filename-to-html($orig-file-name)" />
```

with just:
```xslt
<xsl:variable name="petal-source-file" select="com:filename(com:document-uri(.))" />
<xsl:variable name="petal-file-name" select="hcom:dita-filename-to-html($petal-source-file)" />
```

Could you also rename `petal-file-name` to something more descriptive please? Perhaps `petal-webpage-filename`.

Reply:
@adamretter Fixed the renaming of the variable with 93e0906.

Regarding the usage of function `hcom:dita-filename-to-html()`:
I am already using it, but the difficulty is to retrieve and pass the correct argument to the function. 
Hence the complex string processing   to get the correct value for `$petal-source-file` that you use in your example above.
 
Btw, your suggestion would return a filename like this `cityehr-quick-start-guide/src/main/lwdita/quickstart-guide-modular/introduction.html`, but we need the `.xdita` file ending (the GitHub URI), therefore my initial processing of the document URI. 

To summarize the required URL parts of example "introduction.html":
`$petal-base-url`  -> `https://petal.evolvedbinary.com/`
`?ghrepo=`  -> `evolvedbinary/cityehr-documentation`
`&source=` -> `cityehr-quick-start-guide/src/main/lwdita/quickstart-guide-modular/introduction.dita`
`&branch=` -> `develop`
`&referer=` -> `https://evolvedbinary.github.io/cityehr-documentation/introduction.html`

Reply:
@alex how about just:

```
<xsl:variable name="petal-source-file-uri" select="com:document-uri(.)" />
          <xsl:variable name="petal-source-file" select="substring-after($petal-source-file-uri, $petal-github-repo-name || '/')" />
          <xsl:variable name="petal-webpage-filename" select="hcom:dita-filename-to-html(com:filename($petal-source-file-uri))" />
          <xsl:variable name="petal-full-url" select="concat($petal-api-url, '?ghrepo=', $petal-github-org-name, '/', $petal-github-repo-name, '&amp;source=', $petal-source-file, '&amp;branch=', $petal-github-branch, '&amp;referer=', $petal-referrer-base-url, $petal-webpage-filename)" />
```

Reply:
Yes, thanks for the hint, that fixes it! Fixed with b3f5cbc.